core/node/node_framework/examples/main_node.rs

```diff
@@ -334,7 +337,17 @@ impl MainNodeBuilder {
     }
 
     fn add_commitment_generator_layer(mut self) -> anyhow::Result<Self> {
-        self.node.add_layer(CommitmentGeneratorLayer);
+        let genesis = GenesisConfig::from_env()?;
+        let input_generator: Box<dyn InputGenerator> = if genesis
+            .l1_batch_commit_data_generator_mode
+            == L1BatchCommitDataGeneratorMode::Validium
+        {
+            Box::new(ValidiumInputGenerator)
+        } else {
+            Box::new(RollupInputGenerator)
+        };
```

Comment:
Strictly speaking, this is not idiomatic for the framework. An idiomatic way would be to have input generator as a resource, or at least pass `L1BatchCommitDataGeneratorMode` to the layer and instantiate the input generator inside (like it's done in the [eth_sender layer](https://github.com/matter-labs/zksync-era/blob/main/core/node/node_framework/src/implementations/layers/eth_sender.rs#L93)).

Overall, it's the layers who are responsible for interaction with concrete objects, and the point of the framework is that you don't do any kind of wiring _outside_ of layers.